src/main/java/gregtech/common/mui/widget/GTFluidSlot.java

```diff
@@ -132,14 +131,14 @@ public void draw(ModularGuiContext context, WidgetSlotTheme widgetTheme) {
             this.textRenderer.setPos(0, 12);
             this.textRenderer.draw(amount);
         }
+    }
 
-        if (isHovering()) {
-            GlStateManager.colorMask(true, true, true, false);
-            GuiDraw.drawRect(1, 1, getArea().w() - 2, getArea().h() - 2, widgetTheme.getSlotHoverColor());
-            GlStateManager.colorMask(true, true, true, true);
+    @Override
+    public void drawOverlay(ModularGuiContext context, WidgetTheme widgetTheme) {
+        super.drawOverlay(context, widgetTheme);
+        if (widgetTheme instanceof WidgetSlotTheme slotTheme) {
+            RenderUtil.handleSlotOverlays(this, slotTheme);
         }
```

Comment:
why move this method call into `drawOverlay()` instead of `draw(ModularGuiContext, WidgetSlotTheme)`? it'll render above the tooltip like this

same also goes for our other slot widgets